www/docs/getting-started/upgrade-to-v4.md

```diff
@@ -0,0 +1,155 @@
+- `options`: Not exposed anymore, use `SessionProvider` props
+- `session`: Rename to `getSession`
+- `providers`: Rename to `getProviders`
+- `csrfToken`: Rename to `getCsrfToken`
+- `signin`: Rename to `signIn`
+- `signout`: Rename to `signOut`
+- `Provider`: Rename to `SessionProvider`
+
+Introduced in https://github.com/nextauthjs/next-auth/releases/tag/v4.0.0-next.12
+
+### 3. SessionProvider
+
+Version 4.0 makes using the `SessionProvider` mandatory. This means that you will have to wrap any part of your application using `useSession` in this provider, if you were not doing so already. The `SessionProvider` has also undergone a few further changes:
+
+- `Provider` is renamed to `SessionProvider`
+- The options prop is now flattened as the props of SessionProvider.
+- `clientMaxAge` has been renamed to `staleTime`.
+- `keepAlive` has been renamed to `refetchInterval`.
+
+The best practice for wrapping your app in Providers is to do so in your `pages/_app.jsx` file.
+
+An example use-case with these new changes:
+
+```jsx
+import { SessionProvider } from "next-auth/react"
+
+export default function App({
+  Component,
+  pageProps: { session, ...pageProps },
+}) {
+  return (
+    // `session` comes from `getServerSideProps` or `getInitialProps`.
+    // Avoids flickering/session loading on first load.
+    <SessionProvider session={session}>
+      <Component {...pageProps} />
+    </SessionProvider>
+  )
+}
+```
+
+Introduced in https://github.com/nextauthjs/next-auth/releases/tag/v4.0.0-next.12
+
+### 4. Named Parameters
+
+We have changed the arguments to our callbacks to the named parameters pattern. This way you don't have to use dummy `_` placeholders or other tricks.
+
+The signatures for the callback methods now look like this:
+
+```diff
+- signIn(user, account, profileOrEmailOrCredentials)
++ signIn({ user, account, profile, email, credentials })
+```
+
+```diff
+- redirect(url, baseUrl)
++ redirect({ url, baseUrl })
+```
+
+```diff
+- session(session, tokenOrUser)
++ session({ session, token, user })
+```
+
+```diff
+- jwt(token, user, account, OAuthProfile, isNewUser)
++ jwt({ token, user, account, profile, isNewUser })
+```
+
+Introduced in https://github.com/nextauthjs/next-auth/releases/tag/v4.0.0-next.17
+
+### 5. useSession Hook
+
+The `useSession` hook has been updated to return an object. This allows you to test states much more cleanly with the new `status` option.
+
+```diff
+- const [ session, loading ] = useSession()
++ const { data: session, status } = useSession()
++ const loading = status === "loading"
+```
+
+Introduced in https://github.com/nextauthjs/next-auth/releases/tag/v4.0.0-next.18
+
+### 6. nodemailer
+
+`nodemailer` is no longer a dependency and added by default. If you are using the Email provider you can install it in your project manually, or use any other Email library in the [`sendVerificationRequest`](/configuration/providers#options-1#:~:text=sendVerificationRequest) callback.
```

Comment:
```suggestion
[`nodemailer`](https://www.npmjs.com/package/normalizer) is no longer a dependency and added by default. If you are using the Email provider you can install it in your project manually, or use any other Email library in the [`sendVerificationRequest`](/configuration/providers#options-1#:~:text=sendVerificationRequest) callback.
```

Reply:
Small copy + paste error I think, but good call adding the link. Ill commit that myself :+1: